pipelines/pipelines/tensorflow/prediction/queries/ingest.sql

```diff
@@ -4,24 +4,31 @@
 -- you may not use this file except in compliance with the License.
 -- You may obtain a copy of the License at
 
---     https://www.apache.org/licenses/LICENSE-2.0
+--      http://www.apache.org/licenses/LICENSE-2.0
 
 -- Unless required by applicable law or agreed to in writing, software
 -- distributed under the License is distributed on an "AS IS" BASIS,
 -- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 -- See the License for the specific language governing permissions and
 -- limitations under the License.
 
-with filtered_data as (
+-- Treat "filter_start_value" as the current time, unless it is empty then use CURRENT_DATETIME() instead
+-- This allows us to set the filter_start_value to a specific time for testing or for backfill
+with filter_start_values as (
+    SELECT 
+    IF("{{ filter_start_value }}" = '', CURRENT_DATETIME(), CAST("{{ filter_start_value }}" AS DATETIME)) as filter_start_value
```

Comment:
I prefer coalesce but that's personal preference, though it can deal with null values, too (not just empty strings).

Reply:
I think I tried that, but I couldn't get it working

`CURRENT_DATETIME()` is type DATETIME, whereas `filter_start_value` will be a string. You need to cast `filter_start_value` to a DATETIME, but if it's empty/null you will get an error. Can you think of a way to make it work with `COALESCE`? I agree it would be neater

Reply:
You could still make it work, if you ensure that `filter_start_value` is parsed as `NULL` if empty before calling `COALESCE`. However, I don't see this resulting in cleaner or more robust code. Feel free to resolve this then!